br/pkg/lightning/backend/external/merge.go

```diff
@@ -76,6 +76,22 @@ func MergeOverlappingFiles(
 
 // mergeOverlappingFilesInternal reads from given files whose key range may overlap
 // and writes to one new sorted, nonoverlapping files.
+// since some memory are taken by library, such as HTTP2, that we cannot calculate
+// accurately, here we only consider the memory used by our code, the estimate max
+// memory usage of this function is:
+//
+//	memSizeLimit
+//	+ 20 * partSize
+//	+ 20 * 5MiB(stat file, we might not use all part, as stat file is quite small)
+//	+ readBufferSize * len(paths)
```

Comment:
in NewMergeKVIter, there's an extra memory consumption of  concurrenct reading

Reply:
https://github.com/pingcap/tidb/pull/51486/commits/825d84cb8b745056578c7e8a81eb9202d16725c8